drivers/mspi/mspi_stm32_ospi.c

```diff
@@ -0,0 +1,1838 @@
+			return -EIO;
+		}
+	}
+
+	if (IS_ENABLED(MSPI_STM32_DOMAIN_CLOCK_SUPPORT) && (cfg->pclk_len > 2)) {
+		if (clock_control_off(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+				      (clock_control_subsys_t)&cfg->pclken[2]) != 0) {
+			LOG_ERR("Could not enable XSPI Manager clock");
+			return -EIO;
+		}
+	}
+
+	/* Move pins to sleep state */
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_SLEEP);
+	if (ret == -ENOENT) {
+		/* Warn but don't block suspend */
+		LOG_WRN_ONCE("MSPI pinctrl sleep state not available");
+		return 0;
+	}
+
+	return ret;
+}
+
+static int mspi_stm32_ospi_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		return mspi_stm32_ospi_activate(dev);
+	case PM_DEVICE_ACTION_SUSPEND:
+		return mspi_stm32_ospi_suspend(dev);
+	default:
+		return -ENOTSUP;
+	}
+}
+#endif /* CONFIG_PM_DEVICE */
+
+#define DMA_CHANNEL_CONFIG(node, dir) DT_DMAS_CELL_BY_NAME(node, dir, channel_config)
+
+#define OSPI_DMA_CHANNEL_INIT(node, dir)                                                           \
+	.dev = DEVICE_DT_GET(DT_DMAS_CTLR(node)),                                                  \
+	.channel = DT_DMAS_CELL_BY_NAME(node, dir, channel),                                       \
+	.reg = (DMA_TypeDef *)DT_REG_ADDR(DT_PHANDLE_BY_NAME(node, dmas, dir)),                    \
+	.cfg = {                                                                                   \
+		.dma_slot = DT_DMAS_CELL_BY_NAME(node, dir, slot),                                 \
+		.source_data_size =                                                                \
+			STM32_DMA_CONFIG_PERIPHERAL_DATA_SIZE(DMA_CHANNEL_CONFIG(node, dir)),      \
+		.dest_data_size =                                                                  \
+			STM32_DMA_CONFIG_MEMORY_DATA_SIZE(DMA_CHANNEL_CONFIG(node, dir)),          \
+		.channel_priority = STM32_DMA_CONFIG_PRIORITY(DMA_CHANNEL_CONFIG(node, dir)),      \
+		.dma_callback = mspi_stm32_ospi_dma_callback,                                      \
+	},
+
+#define OSPI_DMA_CHANNEL(node, dir)                                                                \
+	.dma = {COND_CODE_1(DT_DMAS_HAS_NAME(node, dir),     \
+			(OSPI_DMA_CHANNEL_INIT(node, dir)),     \
+			(NULL)) },
+
+/* MSPI control config */
+#define MSPI_CONFIG(n)                                                                             \
+	{                                                                                          \
+		.channel_num = 0,                                                                  \
+		.op_mode = DT_ENUM_IDX_OR(n, op_mode, MSPI_OP_MODE_CONTROLLER),                    \
+		.duplex = DT_ENUM_IDX_OR(n, duplex, MSPI_HALF_DUPLEX),                             \
+		.max_freq = DT_INST_PROP_OR(n, mspi_max_frequency, MSPI_STM32_MAX_FREQ),           \
+		.dqs_support = DT_INST_PROP_OR(n, dqs_support, false),                             \
```

Comment:
For boolean DT properties, ` = DT_INST_PROP(n, <prop-name>)` is enough.
Ditto for `.sw_multi_periph`.

Comments also apply to QSPI and XSPI.